exercises/express-notes-api/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+
+const app = express();
+
+const fs = require('fs');
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Listening on port 3000!');
+});
+
+app.get('/api/notes', (req, res) => {
+  const notesData = [];
+  fs.readFile('data.json', 'utf-8', (err, data) => {
+    if (err) {
+      console.error(err);
+      res.status(500).send({ error: 'an unexpected error occured' });
+      return;
+    }
+    const parsedData = JSON.parse(data);
+    for (const key in parsedData.notes) {
+      notesData.push(parsedData.notes[key]);
+    }
+    res.status(200).send(notesData);
+  });
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  fs.readFile('data.json', 'utf-8', (err, data) => {
+    if (err) {
+      console.error(err);
+      res.status(500).send({ error: 'an unexpected error occured' });
+      return;
+    }
+    const parsedData = JSON.parse(data);
+    if (Math.sign(req.params.id) === 1 && Number.isInteger(parseFloat(req.params.id))) {
+      if (parsedData.notes[req.params.id]) {
+        res.status(200).send(parsedData.notes[req.params.id]);
+      } else {
+        res.status(404).send({ error: `cannot find not with id ${req.params.id}` });
+      }
+    } else {
+      res.status(400).send({ error: 'id must be a positive integer' });
+    }
+  });
+});
+
+app.post('/api/notes', (req, res) => {
+  fs.readFile('data.json', 'utf-8', (err, data) => {
+    if (err) {
+      console.error(err);
+      res.status(500).send({ error: 'an unexpected error occured' });
+      return;
+    }
+    const parsedData = JSON.parse(data);
+    if (req.body.content) {
```

Comment:
Same here, it's better to check this first and bail out before reading the file.